gtsfm/loader/one_d_sfm_loader.py

```diff
@@ -51,18 +52,45 @@ def __init__(
                 the smaller of the height/width of the image. e.g. for 1080p (1920 x 1080),
                 max_resolution would be 1080. If the image resolution max(height, width) is
                 greater than the max_resolution, it will be downsampled to match the max_resolution.
+            enable_no_exif: flag to whether to read images without exif.
         """
         super().__init__(max_resolution=max_resolution)
+        self._enable_no_exif = enable_no_exif
 
         # Fetch all the file names in /images folder.
         search_path = os.path.join(folder, "images", f"*.{image_extension}")
 
-        self._image_paths = glob.glob(search_path)
-        self._num_imgs = len(self._image_paths)
+        if not self._enable_no_exif:
+            self._image_paths = glob.glob(search_path)
+        else:
+            (self._image_paths, num_all_imgs) = self.get_images_with_exif(search_path)
+            logger.info("Read %d images with exif out of %d in total.", len(self._image_paths), num_all_imgs)
 
+        self._num_imgs = len(self._image_paths)
         if self._num_imgs == 0:
             raise RuntimeError(f"Loader could not find any images with the specified file extension in {search_path}")
 
+    def get_images_with_exif(self, search_path: str) -> Tuple[List[str], int]:
```

Comment:
I think this makes more sense in the base loader (it's not necessarily specific to 1dsfm).

Reply:
Done.